### PR TITLE
ITC-35 | Unit and Integration Test

### PR DESCRIPTION
**Pull request by Rahulgarg30591**
In this PR, the Unit and Integrations tests are introduced.

Kindly Review
Rahul Garg

### CODE REVIEW

**Comment by artursmirnov** [comment]
@Rahulgarg30591 thank you for the test! It looks good to me. Although, could you please either rebase your branch from upstream, or just cherry-pick the commit with the test to a new branch, so that PR contain only relevant changes?

**Comment by Rahulgarg30591** [comment]
Hello @artursmirnov sure, I thought of adding integration tests as well. Now the branch is rebased to the main epic-fast-track branch. I have also performed some code cleaning from my side. Kindly review the same now.

cc: @shairyar

File: tests/integration/components/build-header-test.js [hunk 1]
```diff
@@ -219,4 +219,40 @@ module('Integration | Component | build header', function (hooks) {
...
+      },
+      repo,
+    };
+
+    this.set('build', build);
+    await render(hbs`{{build-header item=build}}`);
+    assert.dom('.build-prioritize').hasText('Prioritized build');
+  });
+
+  test('show prioritize heading', async function (assert) {
```
Comment (artursmirnov) [inline tests/integration/components/build-header-test.js:241]: The name of this test does not relate to its content. Should it be "does not show prioritize heading"?

File: tests/integration/components/build-header-test.js [hunk 2]
```diff
@@ -219,4 +219,40 @@ module('Integration | Component | build header', function (hooks) {
     await render(hbs`{{build-header item=job}}`);
     assert.dom('.detail-job-arch').exists('does display CPU architecture');
   });
+
+  test('show prioritize heading', async function (assert) {
```
Comment (artursmirnov) [inline tests/integration/components/build-header-test.js:223]: Could you please make test titles a little bit more descriptive, e.g. "it shows prioritize heading when build is of high priority"?

Comment (Rahulgarg30591) [reply tests/integration/components/build-header-test.js]: It got missed, sorry now fixed.

File: tests/integration/components/build-header-test.js (hunk 1 above)
Comment (Rahulgarg30591) [reply tests/integration/components/build-header-test.js]: Done.

File: tests/integration/components/repo-actions-test.js [hunk 3]
```diff
@@ -50,4 +49,23 @@ module('Integration | Component | repo actions', function (hooks) {
     await render(hbs`{{repo-actions userHasPullPermissionForRepo=true job=this.job}}`);
     assert.dom('button[aria-label="Restart job"]').exists();
   });
+
+  test('it shows priortize button only if build is not prioritized, it is also not already started and if the org and user are having acces of it', async function (assert) {
```
[the remark this answers is not in the document]
Comment (Rahulgarg30591) [reply tests/integration/components/repo-actions-test.js]: Typos :( Sorry I will check it thoroughly before asking for the review next time. Now fixed.

File: tests/integration/components/templates/prioritize-build-modal-test.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+});
+
+module('Integration | Component | Dialog | prioritize build modal', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    stubService('auth', authStub);
+  });
+
+  test('it shows the priority action model and loading indicator if task is in running state', async function (assert) {
```
[the remark this answers is not in the document]
Comment (Rahulgarg30591) [reply tests/integration/components/templates/prioritize-build-modal-test.js]: Now fixed.

File: app/templates/components/repo-actions.hbs [hunk 5]
```diff
@@ -47,6 +47,31 @@
       </button>
     </div>
   {{/if}}
+  {{#if this.canPrioritize}}
+    <div class="action-button-container">
+      {{#if this.isTooltip}}
```
[the remark this answers is not in the document]
Comment (Rahulgarg30591) [reply app/templates/components/repo-actions.hbs]: insufficientPermissions sounds clean, changed the code accordingly.

File: app/templates/components/repo-actions.hbs [hunk 6]
```diff
@@ -47,6 +47,31 @@
...
+            Prioritize build
+          </EmberTooltip>
+        {{/if}}
+        <SvgImage @name="icon-prioritize" @class="icon prioritize-icon" />
+        <span class="label-align">
+          Prioritize build
+        </span>
+      </button>
+    </div>
+  {{/if}}
```
[the remark this answers is not in the document]
Comment (Rahulgarg30591) [reply app/templates/components/repo-actions.hbs]: Removed.

File: app/templates/components/repo-actions.hbs [hunk 7]
```diff
@@ -96,9 +121,9 @@
     <LoadingIndicator />
   {{/if}}
 {{/if}}
-<Dialogs::PrioritizeBuildModal 
+<Dialogs::PrioritizeBuildModal
   @isOpen={{this.showPrioritizeBuildModal}}
-  @onClose={{fn (mut this.showPrioritizeBuildModal) false}}
+  @onClose={{action this.closePriorityModel}}
```
[the remark this answers is not in the document]
Comment (Rahulgarg30591) [reply app/templates/components/repo-actions.hbs]: Done.

File: app/templates/components/dialogs/prioritize-build-modal.hbs [hunk 8]
```diff
@@ -29,8 +29,8 @@
     >
       <form.field
         @disableFrame={{true}}
-        @value={{this.shouldCancelRunningJobs}}
-        onclick={{action (mut this.shouldCancelRunningJobs) true}}
+        @value={{not this.shouldCancelRunningJobs}}
+        onclick={{action (mut this.shouldCancelRunningJobs) false}}
```
[the remark this answers is not in the document]
Comment (Rahulgarg30591) [reply app/templates/components/dialogs/prioritize-build-modal.hbs:33]: Welcome.

File: app/components/repo-actions.js [hunk 9]
```diff
@@ -79,6 +82,14 @@ export default Component.extend({
 
   restarters: taskGroup().drop(),
 
+  closePriorityModel: function () {
```
[the remark this answers is not in the document]
Comment (Rahulgarg30591) [reply app/components/repo-actions.js]: Done.

**Comment by Rahulgarg30591** [comment]
Hello, @artursmirnov thank you for the review, I have made the required changes. May you kindly review the same again, Thank you.

File: tests/integration/components/repo-actions-test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (artursmirnov) [reply tests/integration/components/repo-actions-test.js]: No worries, this is what code reviews are for ;)

**Comment by shairyar** [comment]
Thank you @artursmirnov for all the help 👍

**Pull request merged**